### ISSUE TITLE
Something wrong with i18n scope & rails 6.0.0.rc2

### ISSUE THREAD

**Issue by jhovad** [shioyama/mobility#334]
```ruby
class Category < ActiveRecord::Base
  extend Mobility
  translates :name, :description, :banner_bottom, :url, :publish, :meta_title, :teaser, :seo_text, :tag_h1
  default_scope { i18n }
  #...
end
```

6.0.0.rc1 - ok:
```ruby
irb(main):002:0> Category.where(publish: true)
D, [2019-08-02T10:25:38.551464 #4822] DEBUG -- :   Category Load (1.0ms)  SELECT "categories".* FROM "categories" INNER JOIN "category_translations" "category_translations_cz" ON "category_translations_cz"."category_id" = "categories"."id" AND "category_translations_cz"."locale" = 'cz' WHERE "category_translations_cz"."publish" = TRUE LIMIT $1  [["LIMIT", 11]]
=> #<ActiveRecord::Relation [#<Category ....
```

6.0.0.rc2 - something strange:
```ruby
irb(main):001:0> Category.where(publish: true)
W, [2019-08-02T10:26:36.672560 #5008]  WARN -- : DEPRECATION WARNING: `Module#parent` has been renamed to `module_parent`. `parent` is deprecated and will be removed in Rails 6.1. (called from <class:Category> at /Users/josef/Documents/Develop/bellinda/app/models/category.rb:11)
Traceback (most recent call last):
        1: from (irb):1
ArgumentError (wrong number of arguments (given 2, expected 1))
```

**Comment by shioyama** [comment]
I'll look into it, but I won't be making any changes to fix release candidates since things can change approaching the first release.

But that deprecation warning looks like it might be related...

**Comment by fauno** [comment]
Got something similar with Rails 6.0.0:

```
ArgumentError: wrong number of arguments (given 2, expected 1)
from /home/fauno/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/mobility-0.8.7/lib/mobility/arel/visitor.rb:17:in `visit'
```

**Comment by ssaunier** [comment]
@fauno I had something like that last week and I think it was related to `pg_search`. Do you have this gem as well in your project?

**Comment by fauno** [comment]
hi! no, i'm using sqlite3 for database, the exception happens when i'm trying to save the model:

```ruby
m = Model.new(mobility_attribute: 'hi')
m.mobility_attribute
=> "hi"
m.valid?
ArgumentError: wrong number of arguments (given 2, expected 1)
from /home/fauno/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/mobility-0.8.7/lib/mobility/arel/visitor.rb:17:in `visit'
```

**Comment by fauno** [comment]
This does the trick:

```diff
--- lib/mobility/arel/visitor.rb.orig	2019-08-29 14:20:24.180122614 -0300
+++ lib/mobility/arel/visitor.rb	2019-08-29 14:18:38.146723362 -0300
@@ -14,7 +14,7 @@
 
       private
 
-      def visit(object)
+      def visit(object, collection = nil)
         super
       rescue TypeError
         visit_default(object)
```

**Comment by abunsen** [comment]
👋🏾 Will this patch be included in a version soon?

**Comment by shioyama** [comment]
The change that broke this is rails/rails@85b4ba28364. Mobility shouldn't  really be overriding private methods, that's the real problem here.

If I merge that change it will break apps using AR < 6.0. The better fix is:

```diff
--- lib/mobility/arel/visitor.rb.orig	2019-08-29 14:20:24.180122614 -0300
+++ lib/mobility/arel/visitor.rb	2019-08-29 14:18:38.146723362 -0300
@@ -14,7 +14,7 @@
 
       private
 
-      def visit(object)
+      def visit(*args)
         super
       rescue TypeError
         visit_default(*args)
```

This way no change in argument count will break the override.

**Comment by shioyama** [comment]
The fix for this was released in 0.8.8.

**Comment by n00dle** [comment]
I know this is an old issue, but it was the only result that came up on my search for a similar error message.  Dropping this here in the hopes that it saves someone else some time.

I was upgrading a rails 5.2 app to rails 6.0 (on the way to something newer). In my case, a dependency included the `arel` gem, which has since been added into ActiveRecord. When AR ran, it picked up the external `arel` gem and spit out the `ArgumentError`.

```
ArgumentError: wrong number of arguments (given 2, expected 1)
gemdir/arel-9.0.0/lib/arel/visitors/to_sql.rb:73:in `compile'
gemdir/activerecord-6.0.6.1/lib/active_record/connection_adapters/abstract/database_statements.rb:35:in `to_sql_and_binds'
```
